File: r-ggplot/chart.R
if(!require(ggplot2)) install.packages("ggplot2")
library(ggplot2)

data <- read.csv('../penglings.csv')

ggplot(data, aes(x=flipper_length_mm, y=body_mass_g, size=bill_length_mm, fill=species)) +

    geom_point(alpha=0.8, shape=21, color="black") +
    scale_fill_manual(values=c('Adelie'='#017339', 'Gentoo'='#e8ac0e', 'Chinstrap'='#e25501')) +
    scale_size(range = c(3, 8)) +

    labs(title='Penguin Size vs Mass (R + ggplot2)',
    x='Flipper Length (mm)',
    y='Body Mass (g)') +

    theme_minimal()

    ggsave('a2-r-ggplot-graph.png', width=6, height=4)

File: altair/chart.py
import altair as alt
import pandas as pd


df = pd.read_csv('../penglings.csv')

chart = alt.Chart(df).mark_circle(opacity=0.8, stroke='black', strokeWidth=1).encode(

x=alt.X('flipper_length_mm', scale = alt.Scale(zero=False), title = 'Flipper Length (mm)'),
y=alt.Y('body_mass_g', scale=alt.Scale(zero=False), title='Body Mass (g)'),

color=alt.Color('species', scale=alt.Scale(
domain=['Adelie', 'Gentoo', 'Chinstrap'],
range=['#017339', '#e8ac0e', '#e25501']
)),

size = alt.Size('bill_length_mm', title='Bill Length (mm)'),

tooltip = ['species', 'flipper_length_mm', 'body_mass_g', 'bill_length_mm']).properties(
title='Penguin Size vs Mass',
width = 600,
height = 480
).interactive()

chart.save('a2-altair-graph.html')
print('Chart saved as a2-altair-graph.html')
